dashboard/app/helpers/aichat_sagemaker_helper.rb

```diff
@@ -3,7 +3,10 @@ module AichatSagemakerHelper
   TOP_P = 0.9
 
   def self.create_sagemaker_client
-    Aws::SageMakerRuntime::Client.new
+    # Stubbed SageMaker allows UI tests (without the roundtrip to the model) to run in CI environments (ie, Drone)
+    Rails.application.config.respond_to?(:stub_aichat_aws_services) && Rails.application.config.stub_aichat_aws_services ?
```

Comment:
I see other instances of checking `Rails.application.config` but without `respond_to`. 
https://github.com/code-dot-org/code-dot-org/blob/d9a0da937c1ca95c2587c9e94c3b20071be02e3a/dashboard/app/controllers/scripts_controller.rb#L159

https://github.com/code-dot-org/code-dot-org/blob/d9a0da937c1ca95c2587c9e94c3b20071be02e3a/dashboard/app/models/unit.rb#L409

Do you mind explaining why you need `respond_to` here? Thanks!

Reply:
mmm good q, I believe it's because those are defined in all environments, whereas this bit of config is defined only in `test` environments. Whether I should be defining this everywhere is another question...cc @cat5inthecradle

![image](https://github.com/user-attachments/assets/65991ed1-dbdb-45c7-9c6b-a4bf4b877941)

Reply:
I'm generally okay with only defining it where it's non-default. I could go either way. I also like being explicit :man_shrugging:

Reply:
Maybe one benefit of having it only set in `test.rb` is that it provides another layer of protection against accidentally setting it to `true` in non-test environments (I have a deep fear I'm about to start sending "Hello there!" as a response to every chat completion request)

Reply:
Good point! Thanks for the discussion.